rust/src/main.rs

```diff
@@ -73,15 +72,62 @@ fn main() -> anyhow::Result<()> {
         }
     }));
 
+    let enabled_flags: HashSet<_> = config.features.file_flags.iter().cloned().collect();
+
     for test_case in TEST_CASES {
-        if let Some(sc) = &test_case.syscall {
-            if !enabled_features.contains(sc) {
-                println!(
-                    "skipped {}: please add it to the configuration file if you want to run it",
-                    sc
-                );
-                continue;
-            }
+        //TODO: There's probably a better way to do this...
+        let mut should_skip = false;
+
+        let mut message = None;
+
+        let features = test_case
+            .required_features
+            .iter()
+            .cloned()
```

Comment:
Are you sure you need the `.cloned()` here on and on lines 75 and 60?

Reply:
It doesn't compile if I remove them.

Reply:
My bad, it was a useless `into_iter` which was causing the issue.